sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLoggingPolicy.java

```diff
@@ -128,18 +172,25 @@ public HttpLoggingPolicy(HttpLogDetailLevel detailLevel, boolean prettyPrintJSON
                 bodySize = contentLengthString + "-byte";
             }
 
-            //            HttpResponseStatus responseStatus = HttpResponseStatus.valueOf(response.statusCode());
-            if (detailLevel.shouldLogUrl()) {
+            // HttpResponseStatus responseStatus = HttpResponseStatus.valueOf(response.statusCode());
+            if (httpLogOptions.getLogLevel().shouldLogUrl()) {
                 logger.info("<-- {} {} ({} ms, {} body)", response.getStatusCode(), url, tookMs, bodySize);
             }
 
-            if (detailLevel.shouldLogHeaders()) {
+            if (httpLogOptions.getLogLevel().shouldLogHeaders()) {
+                StringBuilder sb = new StringBuilder();
                 for (HttpHeader header : response.getHeaders()) {
-                    logger.info(header.toString());
+                    sb.append(header.getName()).append(":");
+                    if (httpLogOptions.getAllowedHeaderNames().contains(header.getName())) {
+                        sb.append(header.getValue());
+                    } else {
+                        sb.append(REDACTED_PLACEHOLDER);
+                    }
                 }
+                logger.info(sb.toString());
```

Comment:
This header concatenation code should be centralised in this class rather than duplicated for request and response headers.